force-app/main/default/classes/primaryContactSelectExtension.cls

```diff
@@ -0,0 +1,79 @@
+/**
+ * File:        primaryContactSelectExtension
+ * Project:     VsExerciseProjects
+ * Date:        July 19, 2021
+ * Created By:  Dragos Olaru
+ * *************************************************************************
+ * Description:  Extension class for querying and modifying records
+ * *************************************************************************
+ * History:
+ * Date:       			            Modified By:  		           Description:
+ *              July 22, 2021                      Dragos Olaru
+ */
+global with sharing class primaryContactSelectExtension {
+  public Account person;
+  public primaryContactSelectExtension(
+    ApexPages.StandardController stdController
+  ) {
+    this.person = (Account) stdController.getRecord();
+  }
+  public primaryContactSelectExtension() {
+  }
+
+  public list<Contact> contactWorkedOn { get; set; }
+  public String searchKey { get; set; }
+  public String modKey { get; set; }
+
+  /**
+   * @author Dragos Olaru
+   * @Date July 20, 2021
+   * @name search
+   * @description Query for related contacts on account
+   * @return void
+   */
+  public void search() {
+    contactWorkedOn = [
+      SELECT Id, LastName, title, phone, Is_Primary_Contact__c
+      FROM Contact
+      WHERE Accountid = :this.person.Id AND lastname LIKE :'%' + searchKey + '%'
+      LIMIT 10
+    ];
+  }
```

Comment:
You need pagination here to display all results, displaying 10 results at a time will not be enough for real implementation